apps/ejabberd/src/jlib.erl

```diff
@@ -368,22 +363,14 @@ jid_to_binary({Node, Server, Resource}) ->
          end,
     S3.
 
--spec is_nodename([] | binary()) -> boolean().
+-spec is_nodename(<<>> | binary()) -> boolean().
 is_nodename([]) ->
```

Comment:
This clause should be removed if we are sure this function won't be called for lists.